Promo figures for Panorama Public datasets - Updates

vagisha opened the pull request:
#### Rationale
Some updates to the UI for collecting Panorama Public data catalog entries

#### Related Pull Requests
- https://github.com/LabKey/MacCossLabModules/pull/306

#### Changes
- In the success view after data is made public, display a slideshow preview to encourage submitter to provide a catalog entry.
- Added a catalog entry icon to the Targeted MS Experiment webpart. This links to either the action to add a new catalog entry, or view the existing one. It is displayed only in the most recent version of the data in the case that there are multiple copies of a dataset (no such datasets yet on Panorama Public).
- Updated text in the catalog entry form.
- Post message to support thread when a catalog entry is added.
- Added "View Slideshow" button in the Catalog Entry table toolbar. Visible only to site admins. This can used to preview the pending entries, for example.
  - Updated viewSlideshow.jsp - added form elements to choose the catalog entry type (all, approved, rejected, pending) and the number of entries to display.
- Moved JS and CSS from viewSlideshow.jsp to source files so that they can be reused.
- Added test.

vagisha committed:
- Added catalog entry icon to experimentDetails.jsp
- Working on test

vagisha committed:
- Update text on catalogEntryForm.jsp
- Update success view to include a slideshow preview

vagisha committed:
Replace "slideshow catalog" with catalog.

vagisha committed:
- Display a slideshow preview in the success view after making data public to encourage submitter to provide an entry.
- Post notification when a catalog entry is added.
- Display the catalog entry icon and webpart only in the most recent copy of a dataset.
- Added "View Slideshow" button in the Catalog Entry table toolbar. Visible only to site admins.
- Updated viewSlideshow.jsp. Added form elements to choose the catalog entry type (all, approved, rejected, pending) and the number of entries to display.
- Added test.

vagisha committed:
- Updated notification text.
- Added isCurrentVersion() method to ExperimentAnnotationsManager

vagisha committed:
Fix for failing test.

vagisha committed:
Test that catalog entry image is moved to the new copy when a resubmitted data is copied.

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/query/CatalogEntryManager.java:
```diff
@@ -144,9 +147,10 @@ public static void moveEntry(@NotNull ExperimentAnnotations previousCopy, @NotNu
     }
 
-    public static List<CatalogEntry> getApprovedEntries(int entryCount)
+    public static List<CatalogEntry> getEntries(@NotNull CatalogEntryType entryType, int entryCount)
```
Does entryCount just cap the number of rows it should return? If so, I think you could call `setMaxRows(entryCount)` on the `TableSelector`

vagisha replied:
Yes, this could be simplified with `setMaxRows(entryCount)` and `setOffset(offset)`.  I will do that in the next iteration.

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java:
```diff
@@ -543,6 +543,20 @@ public static boolean hasProteomicData(ExperimentAnnotations experimentAnnotatio
+        if (experimentAnnotations != null && experimentAnnotations.isJournalCopy())
+        {
+            Integer maxVersion = ExperimentAnnotationsManager.getMaxVersionForExperiment(experimentAnnotations.getSourceExperimentId());
+            return experimentAnnotations.getDataVersion().equals(maxVersion);
```
Is `getDataVersion()` guaranteed to be non-null? If so, should it return an `int` instead of `Integer`?

vagisha replied:
It is non-null only for experiments that are copies in the Panorama Public project (isJournalCopy()).

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/view/expannotations/experimentDetails.jsp:
```diff
@@ -191,6 +212,19 @@
+ }
+ a.catalogIconGreen
+ {
+     background: url("../../PanoramaPublic/images/slideshow-icon-green.png") no-repeat bottom 0 right 0;
```
Will this always be the right path? I worry about the `../../` part. Might be better to make it something like:

```suggestion
     background: url("<%= h(contextPath) %>/PanoramaPublic/images/slideshow-icon-green.png") no-repeat bottom 0 right 0;
```

vagisha replied:
Done.

vagisha committed:
- CR feedback - replace ../../ with h(contextPath)
- In AssignSubmitterPermissionJob, assign PanoramaPublicSubmitterRole to form submitter in addition to users set as the submitter and lab head.
